src/Raven.Server/Documents/PeriodicBackup/Azure/RavenAzureClient.cs

```diff
@@ -480,21 +483,174 @@ List<BlobProperties> GetResult()
                         .Select(x => new BlobProperties
                         {
                             Name = x
-                        })
-                        .ToList();
+                        });
                 }
-                else
+
+                return listBlobsResult
+                    .Descendants("Blob")
+                    .Select(x => new BlobProperties
+                    {
+                        Name = x.Element("Name")?.Value,
+                    });
+            }
+        }
+
+        public async Task DeleteMultipleBlobs(List<string> blobs)
+        {
+            if (blobs.Count == 0)
+                return;
+
+            const string xMsDate = "x-ms-date";
+            const string xMsClientRequestId = "x-ms-client-request-id";
+            const string xMsReturnClientRequestId = "x-ms-return-client-request-id";
+
+            var now = SystemTime.UtcNow.ToString("R");
+            var url = $"{GetBaseServerUrl()}/?comp=batch";
+
+            var requestMessage = new HttpRequestMessage(HttpMethods.Post, url)
+            {
+                Headers =
                 {
-                    return listBlobsResult
-                        .Descendants("Blob")
-                        .Select(x => new BlobProperties
+                    { xMsDate, SystemTime.UtcNow.ToString("R") }, { "x-ms-version", AzureStorageVersion }
```

Comment:
shouldn't we use `now` local variable here?

Reply:
done